src/main/kotlin/com/dprint/actions/OnSaveAction.kt

```diff
@@ -0,0 +1,32 @@
+package com.dprint.actions
+
+import com.dprint.config.ProjectConfiguration
+import com.dprint.core.Bundle
+import com.dprint.core.LogUtils
+import com.dprint.services.FormatterService
+import com.intellij.ide.actionsOnSave.impl.ActionsOnSaveFileDocumentManagerListener
+import com.intellij.openapi.components.service
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.project.Project
+
+private val LOGGER = logger<OnSaveAction>()
+
+class OnSaveAction : ActionsOnSaveFileDocumentManagerListener.ActionOnSave() {
```

Comment:
This extension wasn't around when I initially built the plugin. Now instead of having that lower level listener I deleted this should only be triggered when a user actually initiates a save. I believe the `BulkFileListener` listens to actual virtual file saves, which the external formatter would have to trigger before running a format to ensure the format result is undoable.